pyproject.toml

```diff
@@ -49,7 +49,7 @@ dependencies = [
     "rich>=13.0.0",  # databinder
     "aiofile",  # compatible versions controlled through miniopy-async
     "make-it-sync",  # compatible versions controlled through func_adl
-    "ruamel.yaml>=0.18",
+    "ruamel.yaml>=0.18,<0.18.7",  # FIXME: ruamel-yaml v0.18.7 has a breaking API change relative to v0.18.6
```

Comment:
I'm selecting a **temporary** upper bound here instead of using the disallow syntax

```
"ruamel.yaml>=0.18,!=0.18.7,!=0.18.8,!=0.18.9,!=0.18.10
```

as the maintainer is on a [release streak](https://sourceforge.net/p/ruamel-yaml/code/ref/default/tags/) (maybe a bad sign after a year of nothing) and it is clear that there's an API breaking change that is going to have to get resolved, so a this is known to fail and it isn't going to get better with an upstream fix, we should cap for now.